src/cudamatrix/cu-sparse-matrix.h

```diff
@@ -95,6 +95,10 @@ class CuSparseMatrix {
   template <typename OtherReal>
   void CopyToSmat(SparseMatrix<OtherReal> *smat) const;
 
+  /// Copy elements to CuVector. It is the caller's responsibility to resize
+  /// <*vec>.
+  void CopyElementsToVec(CuVector<Real> *vec) const;
```

Comment:
That should be changed yes, I fixed that.

Also created an issue for cublas_copy.

Guoguo

On Tue, Jan 5, 2016 at 4:22 PM, Daniel Povey <notifications@github.com>
wrote:

> In src/cudamatrix/cu-sparse-matrix.h
> <https://github.com/kaldi-asr/kaldi/pull/434#discussion_r48914937>:
>
> > @@ -95,6 +95,10 @@ class CuSparseMatrix {
> >    template <typename OtherReal>
> >    void CopyToSmat(SparseMatrix<OtherReal> *smat) const;
> >
> > +  /// Copy elements to CuVector. It is the caller's responsibility to resize
> > +  /// <*vec>.
> > +  void CopyElementsToVec(CuVector<Real> *vec) const;
>
> should be CuVectorBase
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/434/files#r48914937>.
>